op-supervisor/supervisor/backend/db/sync/server.go

```diff
@@ -0,0 +1,144 @@
+package sync
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// Server handles sync requests
+type Server struct {
+	config      Config
+	validChains map[types.ChainID]struct{}
+}
+
+// NewServer creates a new Server with the given config.
+func NewServer(config Config) (*Server, error) {
+	// Convert root to absolute path for security
+	root, err := filepath.Abs(config.DataDir)
+	if err != nil {
+		return nil, fmt.Errorf("invalid root directory: %w", err)
+	}
+
+	// Verify root directory exists and is actually a directory
+	rootInfo, err := os.Stat(root)
+	if err != nil {
+		return nil, fmt.Errorf("cannot access root directory: %w", err)
+	}
+	if !rootInfo.IsDir() {
+		return nil, fmt.Errorf("root path is not a directory: %s", root)
+	}
+
+	// Build map of valid chains for efficient lookup
+	validChains := make(map[types.ChainID]struct{}, len(config.Chains))
+	for _, chain := range config.Chains {
+		validChains[chain] = struct{}{}
+	}
```

Comment:
The only production call to `NewServer` I see uses a fresh copy of a config without populating Chains, so there would be no validChains if running from `supervisor/service.go`

Reply:
Going to handle this issue in the [op-supervisor/supervisor/service.go](https://github.com/ethereum-optimism/optimism/pull/13357/files#diff-cd8e10c2ce9a330798cbac5f50d68bb00de5d473daa2ab618e92b7111b8fe4bb) comment